Config load error handling doesn't mention the file path

```
go run ./main.go
2020-11-06T20:54:59.837-0800    ERROR   setup   unable to load the config file  {"error": "file could not be read from filesystem"}
github.com/go-logr/zapr.(*zapLogger).Error
        /Users/chris/Code/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
main.main
        /Users/chris/Code/src/sigs.k8s.io/kubebuilder/sample/main.go:61
runtime.main
        /usr/local/Cellar/go/1.15.2/libexec/src/runtime/proc.go:204
exit status 1
make: *** [run] Error 1
```

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/config/config.go#L101

Ref #895

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -98,7 +98,7 @@ func (d *DeferredFileLoader) loadFile() {
 
 	content, err := ioutil.ReadFile(d.path)
 	if err != nil {
-		d.err = fmt.Errorf("file could not be read from filesystem")
+		d.err = fmt.Errorf("could not read file at %s", d.path)
 		return
 	}
 

